src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,297 @@
+                // literal expression
+                // 3. If the argument has variable declarations ('out var' in C#), then insert an additional declaration. Also replace all the
+                // occurence of parameter by using this identifier.
+                // 4. For the rest of the parameters (might include method invocation and different expressions), use the parameter's name to
+                // generate a declaration for them and insert that into caller.
+                var allArguments = syntaxFacts.GetArgumentsOfInvocationExpression(calleeInvocationSyntaxNode);
+                var parameterSymbolAndArguments = allArguments
+                    .Select(argument => (
+                        parameterSymbol: inlineMethodRefactoringProvider.GetParameterSymbol(semanticModel, (TArgumentSyntax)argument,
+                            cancellationToken),
+                        argumentExpression: syntaxFacts.GetExpressionOfArgument(argument)))
+                    .Where(parameterAndArgument =>
+                        parameterAndArgument.parameterSymbol != null
+                        && !parameterAndArgument.parameterSymbol.IsDiscard)
+                    // For params array, it could map to multiple arguments so group it.
+                    .GroupBy(keySelector: parameterAndArgument => parameterAndArgument.parameterSymbol!,
+                        elementSelector: parameterAndArgument => parameterAndArgument.argumentExpression)
+                    .SelectAsArray(grouping => (parameterSymbol: grouping.Key, arguments: grouping.ToImmutableArray()));
+                var allParametersWithArgument = parameterSymbolAndArguments.SelectAsArray(g => g.parameterSymbol);
+
+                // 1. Find all the parameter maps to an identifier from caller. After inlining, this identifier would be used to replace the parameter in callee body.
+                // For params array, it should be included here if it is accept an array identifier as argument.
+                var parametersWithIdentifier = parameterSymbolAndArguments
+                    .Where(parameterAndArguments => ParameterWithIdentifierArgumentFilter(syntaxFacts, semanticModel,
+                        parameterAndArguments, cancellationToken))
+                    .ToImmutableArray();
+
+                // 2. Find all the declaration arguments (e.g. out var declaration in C#).
+                // After inlining, an declaration needs to be put before the invocation. And also use the declared identifier to replace the mapping parameter in callee.
+                // Example:
+                // Before:
+                // void Caller()
+                // {
+                //     Callee(out var x);
+                // }
+                // void Callee(out int i) => i = 100;
+                //
+                // After:
+                // void Caller()
+                // {
+                //     int x;
+                //     x = 100;
```

Comment:
nit: we should consider merging these to just be `int x = 100`.

also, this is a very good comment. so woudl like to see this sort of explanatin elsewhere :)